src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/TypeUpgradeAnalyzer.cs

```diff
@@ -133,14 +133,28 @@ private static void AnalyzeIdentifier(SyntaxNodeAnalysisContext context, IEnumer
                 return;
             }
 
-            // If the identifier resolves to an actual symbol that isn't the old identifier, bail out
-            // Also refrain from adding a diagnostic if the symbol is ambiguous but a candidate symbol matches the
-            // new type. This is useful in cases where the type is referenced as a simple name and the new namespace
-            // has been added but the old namespace hasn't been removed yet.
+            var fullyQualifiedNameNode = context.Node.GetQualifiedName();
+
+            // If the identifier is part of a fully qualified name and the qualified name does not match
+            // the old name, then bail out because it is not referring to that type
+            if ((fullyQualifiedNameNode is CSSyntax.QualifiedNameSyntax || fullyQualifiedNameNode is VBSyntax.QualifiedNameSyntax)
+                && !mapping.OldName.EndsWith(fullyQualifiedNameNode.ToString(), StringComparison.Ordinal))
+            {
+                return;
+            }
+
             var symbolInfo = context.SemanticModel.GetSymbolInfo(context.Node);
 
+            // If the identifier resolves to a symbol that isn't a type symbol (property symbol, for example),
+            // then bail out because this analyzer should only flag type usage.
+            if (symbolInfo.Symbol is ISymbol && symbolInfo.Symbol is not INamedTypeSymbol)
```

Comment:
```suggestion
            if (symbolInfo.Symbol is ISymbol && (!symbolInfo.Symbol is INamedTypeSymbol symbol))
```

would remove the need to do another cast on line 156

Reply:
Good call. Changed.